src/provider.tsx

```diff
@@ -79,6 +82,16 @@ export function AuthKitProvider(props: AuthKitProviderProps) {
   );
 }
 
+// poor-man's "deep equality" check
+function isEqual(a: typeof initialState, b: typeof initialState) {
```

Comment:
we talked about pulling in a proper deep-equality checker here (like lodash.isEqual)... `isEqual` is ~2000 loc (which is about ~10x bigger than this whole module--it felt overkill (but i'm happy to reconsider)).

Reply:
This seems fine for now, but is there some way we can write this so if we add more fields in the future, we'll make sure to consider how it affects this equality check?